onadata/libs/serializers/merged_xform_serializer.py

```diff
@@ -0,0 +1,179 @@
+# -*- coding: utf-8 -*-
+"""
+MergedXFormSerializer class
+"""
+
+import base64
+import json
+import uuid
+
+from django.utils.translation import ugettext as _
+from rest_framework import serializers
+
+from onadata.apps.logger.models import MergedXForm, XForm
+from onadata.apps.logger.models.xform import XFORM_TITLE_LENGTH
+from onadata.libs.utils.common_tags import MULTIPLE_SELECT_TYPE, SELECT_ONE
+from pyxform.builder import create_survey_element_from_dict
+
+
+def _get_fields_set(xform):
+    return [
+        (element.get_abbreviated_xpath(), element.type)
+        for element in xform.survey_elements
+        if element.type not in ['', 'survey']
+    ]
+
+
+def _get_elements(elements, intersect, parent_prefix=None):
+    new_elements = []
+
+    for element in elements:
+        name = element['name']
+        name = name if not parent_prefix else '/'.join([parent_prefix, name])
+        if name in intersect:
+            k = element.copy()
+            if 'children' in element and element['type'] not in [
+                    SELECT_ONE, MULTIPLE_SELECT_TYPE]:
+                k['children'] = _get_elements(
+                    element['children'],
+                    [__ for __ in intersect if __.startswith(name)],
+                    name)
+                if not k['children']:
+                    continue
+            new_elements.append(k)
+
+    return new_elements
+
+
+def get_merged_xform_survey(xforms):
+    """
+    Genertates a new pyxform survey object from the intersection of fields of
+    the xforms being merged.
+    """
+    assert len(xforms) > 1, _("Expecting at least 2 xforms")
+
+    xform_sets = [_get_fields_set(xform) for xform in xforms]
+
+    merged_xform_dict = json.loads(xforms[0].json)
+    children = merged_xform_dict.pop('children')
+    merged_xform_dict['children'] = []
+
+    intersect = set(xform_sets[0]).intersection(*xform_sets[1:])
+    intersect = set([__ for (__, ___) in intersect])
+
+    merged_xform_dict['children'] = _get_elements(children, intersect)
+    del merged_xform_dict['_xpath']
```

Comment:
@ukanga This could be re-written as
```python
...
merged_xform_dict.get("_xpath") and merged_xform_dict.pop("_xpath")
```
On stage and locally, saving a merged dataset triggers the following error
![screen shot 2017-08-21 at 14 19 18](https://user-images.githubusercontent.com/1128227/29517155-60011ea8-867c-11e7-9442-d369c82e04a5.png)